purekit/src/main/kotlin/com/virgilsecurity/purekit/pure/HttpPureClient.java

```diff
@@ -33,214 +33,312 @@
+                new HashMap<>(),
                 this.appToken,
                 PurekitProtosV3Storage.UserRecord.parser()
         );
     }
 
     /**
-     * Obtains user
-     * @param userIds userIds
-     * @return UserRecords
-     * @throws ProtocolHttpException FIXME
-     * @throws ProtocolException FIXME
+     * Obtains user.
+     *
+     * @param userIds User Ids.
+     *
+     * @return UserRecords.
+     *
+     * @throws ProtocolException Thrown if an error from the PHE service has been parsed
+     * successfully.
+     * @throws ProtocolHttpException Thrown if an error from the PHE service has NOT been parsed
+     * successfully. Represents a regular HTTP exception with code and message.
      */
-    public PurekitProtosV3Storage.UserRecords getUsers(Collection<String> userIds) throws ProtocolHttpException, ProtocolException {
-        PurekitProtosV3Storage.GetUserRecords getUserRecords = PurekitProtosV3Storage.GetUserRecords.newBuilder().addAllUserIds(userIds).build();
+    public PurekitProtosV3Storage.UserRecords getUsers(Collection<String> userIds) 
+        throws ProtocolHttpException, ProtocolException {
 
-        return this.client.firePost(
+        if (userIds == null) {
+            throw new NullArgumentException("userIds");
+        }
+//        if (userIds.isEmpty()) { // FIXME can we pass empty userIds?
+//            throw new EmptyArgumentException("userIds");
+//        }
+        
+        PurekitProtosV3Storage.GetUserRecords getUserRecords = 
+            PurekitProtosV3Storage.GetUserRecords.newBuilder().addAllUserIds(userIds).build();
+
+        return client.firePost(
                 getUserRecords,
                 HttpClientProtobuf.AvailableRequests.GET_USERS.getType(),
-                new LinkedHashMap<>(),
+                new HashMap<>(),
                 this.appToken,
                 PurekitProtosV3Storage.UserRecords.parser()
         );
     }
 
     /**
-     * Deletes user
-     * @param userId userIds
-     * @param cascade deletes all user cell keys if true
-     * @throws ProtocolHttpException FIXME
-     * @throws ProtocolException FIXME
+     * Deletes user.
+     *
+     * @param userId User Ids.
+     * @param cascade Deletes all user cell keys if true.
+     *
+     * @throws ProtocolException Thrown if an error from the PHE service has been parsed
+     * successfully.
+     * @throws ProtocolHttpException Thrown if an error from the PHE service has NOT been parsed
+     * successfully. Represents a regular HTTP exception with code and message.
      */
-    public void deleteUser(String userId, boolean cascade) throws ProtocolHttpException, ProtocolException {
-        // TODO: parameters ideally should not be added directly to url string
+    public void deleteUser(String userId, boolean cascade) 
+        throws ProtocolHttpException, ProtocolException {
+        // TODO: parameters ideally should not be added directly to url string -> explain please
```

Comment:
POST has body, while get is just a call to endpoint with query in url?

Reply:
POST/PUT encode parameters to body, GET/DELETE encode parameters to url, but the idea is the same. I don't understand why in case of POST and PUT we pass parameters to firePost/firePut and in case of GET and DELETE we should handle it ourselves before calling fireGet/fireDelete